shadows/framework/src/main/java/org/robolectric/shadows/ShadowBiometricManager.java

```diff
@@ -37,23 +34,60 @@ public class ShadowBiometricManager {
   protected int canAuthenticate() {
     if (RuntimeEnvironment.getApiLevel() >= R) {
       return reflector(BiometricManagerReflector.class, realBiometricManager).canAuthenticate();
-    } else if (biometricServiceConnected) {
-      return BIOMETRIC_SUCCESS;
     } else {
-      boolean hasBiomatrics =
-          ReflectionHelpers.callStaticMethod(
-              BiometricManager.class,
-              "hasBiometrics",
-              ClassParameter.from(
-                  Context.class, RuntimeEnvironment.getApplication().getApplicationContext()));
-      if (!hasBiomatrics) {
-        return BIOMETRIC_ERROR_NO_HARDWARE;
+      int biometricResult =
+          canAuthenticateInternal(0, BiometricManager.Authenticators.BIOMETRIC_WEAK);
+      if (biometricServiceConnected) {
+        return BiometricManager.BIOMETRIC_SUCCESS;
+      } else if (biometricResult != BiometricManager.BIOMETRIC_SUCCESS) {
+        return biometricResult;
       } else {
-        return BIOMETRIC_ERROR_HW_UNAVAILABLE;
+        boolean hasBiometrics =
+            ReflectionHelpers.callStaticMethod(
+                BiometricManager.class,
+                "hasBiometrics",
+                ReflectionHelpers.ClassParameter.from(
+                    Context.class, RuntimeEnvironment.getApplication().getApplicationContext()));
+        if (!hasBiometrics) {
+          return BiometricManager.BIOMETRIC_ERROR_NO_HARDWARE;
+        } else {
+          return BiometricManager.BIOMETRIC_ERROR_HW_UNAVAILABLE;
+        }
       }
     }
   }
 
+  @RequiresPermission(USE_BIOMETRIC)
+  @Implementation
+  protected int canAuthenticate(int authenticators) {
```

Comment:
I am seeing the error:

error: No such method in android.hardware.biometrics.BiometricManager for SDK 29
  protected int canAuthenticate(int authenticators) {

Is this minSdk = R as well?
                ^

Reply:
Thanks for reviewing. @charlesng Could you rebase and check this problem?

Reply:
Yes this method required minSdk = 30 as well, https://developer.android.com/reference/android/hardware/biometrics/BiometricManager#canAuthenticate(int) 

I have updated it. Thanks for the review.

Btw, could I ask how did you see the error? As I cannot see error before in both Android Studio and CI 🙏

Reply:
It's done with the `sdkCheckMode` param:

https://github.com/robolectric/robolectric/blob/master/shadows/framework/build.gradle#L11

For some reason it's either not working in gradle or it's not causing a compile-time error.